File: examples/eApi v1.9/java/eAPI-v19-connector/src/main/java/cz/monetplus/mips/eapi/v19/connector/entity/OneclickInitRequest.java
package cz.monetplus.mips.eapi.v19.connector.entity;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonInclude;
import cz.monetplus.mips.eapi.v19.connector.entity.ext.Extension;
import lombok.*;

@EqualsAndHashCode(callSuper = false)
@Data
@JsonInclude(value = JsonInclude.Include.NON_NULL)
@RequiredArgsConstructor @AllArgsConstructor
public class OneclickInitRequest extends SignBase  {
	@NonNull
	private String merchantId;
	@NonNull
	private String origPayId;
	@NonNull
	private String orderNo;
	private String payMethod;
	private String clientIp;
	private Long totalAmount;
	private String currency;
	private Boolean closePayment;
	@NonNull
	private String returnUrl;
	@NonNull
	private String returnMethod;
	private Customer customer;
	private Order order;
	private Boolean clientInitiated;
	private Boolean sdkUsed;
	private String merchantData;
	private List<Extension> extensions;
   	private Integer ttlSec;

	@Override
	public String toSign() {
		StringBuilder sb = new StringBuilder();
		add(sb, merchantId);
		add(sb, origPayId);
		add(sb, orderNo);
		add(sb, dttm);
		add(sb, clientIp);
		add(sb, totalAmount);
		add(sb, currency);
		add(sb, closePayment);
		add(sb, returnUrl);
		add(sb, returnMethod);
		if (null != customer) add(sb, customer.toSign());
		if (null != order) add(sb, order.toSign());
		add(sb, clientInitiated);
		add(sb, sdkUsed);
		add(sb, merchantData);
                add(sb, ttlSec);
		deleteLast(sb);
		return sb.toString();
	}
}

File: examples/eApi v1.9/dotNET/CsobGTWDemo/CsobGatewayClientExample/Communication/DataObjects/OneclickInitRequest.cs
using System.Text;
using CsobGatewayClientExample.Communication.DataObjects.Ext;
using Newtonsoft.Json;

namespace CsobGatewayClientExample.Communication.DataObjects;

public class OneclickInitRequest : SignBaseRequest
{
    [JsonProperty("origPayId")] public string? OrigPayId { get; set; }

    [JsonProperty("orderNo")] public string? OrderNo { get; set; }

    [JsonProperty("totalAmount")] public long? TotalAmount { get; set; }

    [JsonProperty("currency")] public string? Currency { get; set; }

    [JsonProperty("clientIp")] public string? ClientIp { get; set; }


    [JsonProperty("closePayment")] public bool? ClosePayment { get; set; }

    [JsonProperty("returnUrl")] public string? ReturnUrl { get; set; }

    [JsonProperty("returnMethod")] public string? ReturnMethod { get; set; }

    [JsonProperty("customer")] public Customer? Customer { get; set; }

    [JsonProperty("order")] public Order? Order { get; set; }

    [JsonProperty("clientInitiated")] public bool? ClientInitiated { get; set; }

    [JsonProperty("sdkUsed")] public bool? SdkUsed { get; set; }

    [JsonProperty("merchantData")] public string? MerchantData { get; set; }

    [JsonProperty("extension")] private List<Extension>? Extensions { get; set; }

    [JsonProperty("ttlSec")] public int? TtlSec { get; set; }

    public override string ToSign()
    {
        var sb = new StringBuilder();
        Add(sb, MerchantId);
        Add(sb, OrigPayId);
        Add(sb, OrderNo);
        Add(sb, Dttm);
        Add(sb, ClientIp);
        Add(sb, TotalAmount);
        Add(sb, Currency);
        Add(sb, ClosePayment);
        Add(sb, ReturnUrl);
        Add(sb, ReturnMethod);
        if (null != Customer) Add(sb, Customer.ToSign());
        if (null != Order) Add(sb, Order.ToSign());
        Add(sb, ClientInitiated);
        Add(sb, SdkUsed);
        Add(sb, MerchantData);
	Add(sb, TtlSec);
        DeleteLast(sb);
        return sb.ToString();
    }
}